topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|`min.insync.replicas`
+|===
+--
+
+Retention time::
++
+--
+Retention time is the amount of time that messages are retained in a topic before they are deleted. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`604800000 ms` (7 days)
+
+h|Supported values
+| `milliseconds`, `seconds`, `days`, `months`, `years`, `Unlimited`
+
+h|Kafka property name
+|`retention.ms`
+|===
+--
+
+Retention size::
++
+--
+Retention size is the maximum total size of all log segments in a partition before old log segments are deleted to free up space. By default, no retention size limit is applied, only a retention time limit. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`Unlimited`
+
+h|Supported values
+| `bytes`, `kilobytes`, `megabytes`, `gigabytes`, `terabytes`, `Unlimited`
+
+h|Kafka property name
+|`retention.bytes`
+|===
+--
+
+=== Messages
+
+These properties control how your messages are handled in the Kafka instance.
+
+Maximum message bytes::
++
+--
+Maximum message bytes is the maximum record batch size.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 MB`
+
+h|Kafka property name
+|`max.message.bytes`
+|===
+--
+
+Message timestamp type::
++
+--
+Message timestamp type determines whether the timestamp is generated when the message is created (`CreateTime`) or when the message is appended to the log (`LogAppendTime`).
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`CreateTime`
+
+h|Kafka property name
+|`message.timestamp.type`
+|===
+--
+
+Maximum message timestamp difference::
++
+--
+Maximum message timestamp difference is the maximum difference allowed between the timestamp specified in the message when it leaves the producer and the timestamp recorded when a broker receives the message.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`9223372036854775807 ms` (292271023 years, 2 weeks, 2 days, 13 hours, 12 minutes, 56.064 seconds)
+
+h|Kafka property name
+|`message.timestamp.difference.max.ms`
+|===
+--
+
+Compression type::
++
+--
+Compression type determines the final compression for the topic. The `Producer` compression type retains the original compression set by the producer.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`Producer`
+
+h|Kafka property name
+|`compression.type`
+|===
+--
+
+
+Message format version::
++
+--
+Message format version is the `ApiVersion` value that the broker uses to append messages to topics. This value must be a valid `ApiVersion` value, such as 0.8.2, 0.9.0.0, or 0.10.0.
```

Comment:
Should we use later versions rather than _0.8.2, 0.9.0.0, or 0.10.0_ as examples?

Reply:
Well, it's illustrating the value format, not necessarily the version itself. Note how the second is a 4-digit example.